lib/truffle/java/java_proxy.rb

```diff
@@ -25,5 +28,16 @@ def to_java_object
 
   def eql?(another)
   end
-  
+
+  def self.[](*dims)
+    ArrayJavaProxyCreator.new(java_class, *dims)
+  end
+
+  def self.inherited(thing)
+    thing.__send__(:define_singleton_method, :inherited, method(:inherited))
+    unless Thread.current[:MAKE_PROXY]
+      p "Creating #{thing} inherited from #{self} outside the normal make_proxy mechanism."
```

Comment:
If it hangs around for more than a couple of months we should convert it to use proper logging.